src/backend/embedding/extract_file/extract_file.py

```diff
@@ -430,9 +440,97 @@ def extract_docx_content(self, customer_guid: str, file_path: str, filename: str
+                            text = paragraph.text.strip()
+                            if text:
+                                slide_elements.append({
+                                    "type": "text",
+                                    "content": text
+                                })
+
+                    # Extract table data
+                    if shape.has_table:
+                        table = shape.table
+                        table_data = [
+                            [cell.text.strip() for cell in row.cells]
+                            for row in table.rows
+                        ]
+                        table_text = self.format_table_as_text(table_data)
+                        slide_elements.append({
+                            "type": "table",
+                            "content": table_text
+                        })
+
+                    # Extract images
+                    if shape.shape_type == 13:  # 13 indicates a picture
+                        image = shape.image
+                        image_data = image.blob
+                        image_obj = Image.open(io.BytesIO(image_data))
+                        ocr_text = pytesseract.image_to_string(image_obj).strip()
+                        slide_elements.append({
+                            "type": "image",
+                            "content": ocr_text
+                        })
+                        
+                    if shape.shape_type == MSO_SHAPE_TYPE.CHART:
+                        chart = shape.chart
+                        table_data = []
+                        x_axis_labels = [category for category in chart.plots[0].categories]
+                        first_column_name = chart.series[0].name if chart.series and chart.series[0].name else None
+                        series_names = []
+
+                        for series in chart.series:
+                            series_name = series.name if series.name else "Unnamed Series"
+                            series_names.append(series_name)
+                        table_data.append([first_column_name] + series_names)
+                        for idx, category in enumerate(x_axis_labels):
+                            row = [category]
+                            for series in chart.series:
+                                value = series.values[idx] if idx < len(series.values) else None
+                                row.append(value)
+                            table_data.append(row)
+
+                        # Append the reconstructed table data
+                        slide_elements.append({
+                            "type": "chart_table",
+                            "content": table_data
+                        })   
+
+                # Combine slide elements
+                slide_text = "\n".join(element["content"] for element in slide_elements)
+                results.append({
+                    "metadata": {"slide_number": slide_number},
```

Comment:
make it 
 "metadata": {"page_number": slide_number},
 
 to keep it consistent across all document retrievals